File: app/src/main/java/com/bkjcb/rqapplication/stationCheck/fragment/CheckItemDetailFragment.java
package com.bkjcb.rqapplication.stationCheck.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import com.bkjcb.rqapplication.R;
import com.bkjcb.rqapplication.base.fragment.BaseLazyFragment;
import com.bkjcb.rqapplication.stationCheck.model.CheckContentItem;
import com.bkjcb.rqapplication.stationCheck.model.CheckResultItem;
import com.bkjcb.rqapplication.stationCheck.model.CheckResultItem_;

/**
 * Created by DengShuai on 2020/1/7.
 * Description :
 */
public class CheckItemDetailFragment extends BaseLazyFragment {

    private View view;
    private CheckContentItem contentItem;
    private String uid;//项目id
    private CheckResultItem checkResultItem;
    private EditText mItemRecord;
    private int type = 0;//1不可修改

    public void setType(int type) {
        this.type = type;
    }

    public void setContentItem(CheckContentItem contentItem) {
        this.contentItem = contentItem;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public static CheckItemDetailFragment newInstances(CheckContentItem contentItem, String id) {
        CheckItemDetailFragment fragment = new CheckItemDetailFragment();
        fragment.setContentItem(contentItem);
        fragment.setUid(id);
        return fragment;
    }

    public static CheckItemDetailFragment newInstances(CheckContentItem contentItem, String id, int type) {
        CheckItemDetailFragment fragment = new CheckItemDetailFragment();
        fragment.setContentItem(contentItem);
        fragment.setUid(id);
        fragment.setType(type);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (view == null) {
            view = inflater.inflate(R.layout.fragment_check_item_detail, null);
            TextView mItemType = view.findViewById(R.id.item_type);
            TextView mItemContent = view.findViewById(R.id.item_content);
            TextView mItemBasis = view.findViewById(R.id.item_basis);
            TextView mItemSection = view.findViewById(R.id.item_section);
            mItemRecord = view.findViewById(R.id.item_record);
            mItemType.setText(contentItem.getLeibie());
            mItemContent.setText(contentItem.getXuhao() + "、" + contentItem.getJianchaneirong());
            mItemBasis.setText(contentItem.getJianchayiju());
            mItemSection.setText(contentItem.getJianchalanmu().replace("<br/>", "/"));
            if (type == 1) {
                mItemRecord.setEnabled(false);
            }
        }
        return view;
    }


    @Override
    public void onStart() {
        super.onStart();
        checkResultItem = queryLocalItem();
        if (checkResultItem != null) {
            mItemRecord.setText(checkResultItem.jianchajilu);
        } else {
            checkResultItem = new CheckResultItem(uid, contentItem.getId());
        }
    }

    @Override
    public void onStop() {
        super.onStop();
        saveData();
    }

    private CheckResultItem queryLocalItem() {
        return CheckResultItem.getBox().query().equal(CheckResultItem_.jianchaid, uid).and().equal(CheckResultItem_.jianchaxiangid, contentItem.getId()).build().findFirst();
    }

    private void saveData() {

        checkResultItem.jianchajilu = mItemRecord.getText().toString();
        CheckResultItem.getBox().put(checkResultItem);

    }
}

File: app/src/main/java/com/bkjcb/rqapplication/check/fragment/CheckItemDetailFragment.kt
package com.bkjcb.rqapplication.check.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.check.model.CheckContentItem
import com.bkjcb.rqapplication.check.model.CheckResultItem
import com.bkjcb.rqapplication.check.model.CheckResultItem_
import com.bkjcb.rqapplication.base.datebase.DataUtil
import com.bkjcb.rqapplication.base.fragment.BaseLazyFragment
import kotlinx.android.synthetic.main.fragment_check_item_detail.*

/**
 * Created by DengShuai on 2020/1/7.
 * Description :
 */
class CheckItemDetailFragment : BaseLazyFragment() {
    private lateinit var contentItem: CheckContentItem
    private var uid //项目id
            : String? = null
    private var checkResultItem: CheckResultItem? = null
    private var type = 0 //1不可修改
    fun setType(type: Int) {
        this.type = type
    }

    fun setContentItem(contentItem: CheckContentItem) {
        this.contentItem = contentItem
    }

    fun setUid(uid: String?) {
        this.uid = uid
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (rootView == null) {
            rootView = inflater.inflate(R.layout.fragment_check_item_detail, null)
            item_type.text = contentItem.leibie
            item_content.text ="${contentItem.xuhao}、${contentItem.jianchaneirong}"
            item_basis.text = contentItem.jianchayiju
            item_section.text = contentItem.jianchalanmu!!.replace("<br/>", "/")
            if (type == 1) {
                item_record.isEnabled = false
            }
        }
        return view
    }

    override fun onStart() {
        super.onStart()
        checkResultItem = queryLocalItem()
        if (checkResultItem != null) {
            item_record.setText(checkResultItem!!.jianchajilu)
        } else {
            checkResultItem = CheckResultItem(uid!!, contentItem.id!!)
        }
    }

    override fun onStop() {
        super.onStop()
        saveData()
    }

    private fun queryLocalItem(): CheckResultItem? {
        return DataUtil.getCheckResultItemBox().query().equal(CheckResultItem_.jianchaid, uid!!).and().equal(CheckResultItem_.jianchaxiangid, contentItem!!.id!!).build().findFirst()
    }

    private fun saveData() {
        checkResultItem!!.jianchajilu = item_record.text.toString()
        DataUtil.getCheckResultItemBox().put(checkResultItem!!)
    }

    companion object {
        fun newInstances(contentItem: CheckContentItem, id: String?): CheckItemDetailFragment {
            val fragment = CheckItemDetailFragment()
            fragment.setContentItem(contentItem)
            fragment.setUid(id)
            return fragment
        }

        fun newInstances(contentItem: CheckContentItem, id: String?, type: Int): CheckItemDetailFragment {
            val fragment = CheckItemDetailFragment()
            fragment.setContentItem(contentItem)
            fragment.setUid(id)
            fragment.setType(type)
            return fragment
        }
    }
}